source/userguide/auto_annotator.rst

```diff
@@ -0,0 +1,474 @@
+
+In general, detected entities will be represented in the following format:
+
+.. code-block:: python
+
+	entity = { 
+
+		“body”: (substring of sentence), 
+
+		“start”: (start index), 
+
+		“end”: (end index + 1), 
+
+		“dim”: (entity type), 
+
+		“value”: (resolved value, if it exists), 
+
+	}
+
+To restrict the types of entities returned from the :attr:`parse()` method use the :attr:`entity_types` parameter and pass in a list of entities to restrict parsing to. By default, all entities are allowed.
+For example, we can restrict the output of the previous example by doing the following:
+
+
+.. code-block:: python
+	
+	allowed_entites = ["sys_org”, “sys_amount-of-money”, “sys_time”]
+	sentence = "Apple stock went up $10 last monday."
+	sa.parse(sentence=sentence, entity_types=allowed_entities) 
+
+Auto Annotator Configuration
+----------------------------
+
+The :attr:`DEFAULT_AUTO_ANNOTATOR_CONFIG` shown below is the default config for an Annotator.
+A custom config can be included in :attr:`config.py` by duplicating the default config and renaming it to :attr:`AUTO_ANNOTATOR_CONFIG`.
+Alternatively, a custom config dictionary can be passed in directly to :class:`SpacyAnnotator` or any Annotator class upon instantiation.
+
+
+.. code-block:: python
+
+	DEFAULT_AUTO_ANNOTATOR_CONFIG = { 
+
+		"annotator_class": "SpacyAnnotator",
+
+		"overwrite": False, 
+
+		"annotate": [ 
+
+			{ 
+
+				"domains": "*", 
+
+				"intents": "*", 
+
+				"files": "*", 
+
+				"entities": "*", 
+
+			} 
+
+		], 
+
+		"unannotate_supported_entities_only": True, 
+
+		"unannotate": None, 
+	}
+
+Let's take a look at the allowed values for each setting in an Auto Annotator configuration.
+
+
+``'annotator_class'`` (:class:`str`): The class in auto_annotator.py to use for annotation. By default, :class:`SpacyAnnotator` is used. 
+
+``'overwrite'`` (:class:`bool`): Whether new annotations should overwrite existing annotations in the case of a span conflict. False by default. 
+
+``'annotate'`` (:class:`list`): A list of annotation rules where each rule is represented as a dictionary. Each rule must have four keys: :attr:`domains`, :attr:`intents`, :attr:`files`, and :attr:`entities`.
+Annotation rules are combined internally to create Regex patterns to match selected files. :attr:`*` can be used if all possibilities in a section are to be selected. :attr:`|` is used to represent “or”.  
+Let's take a look at an example annotation rule:
+
+.. code-block:: python
+
+	{
+
+		"domains": "faq|salary", 
+
+		"intents": "*", 
+
+		"files": "train.txt|test.txt", 
+
+		"entities": "sys_amount-of-money|sys_time", 
+
+	}
+
+The rule above would annotate all text files named "train" or "test" in the "faq" and "salary" domains. Only sys_amount-of-money and sys_time entities would be annotated.
+
+.. note::
+
+	The order of the annotation rules matters. A rule will overwrite a previous rule if there are conflicts.
```

Comment:
Yes, since the overwrite in the default config refers to whether existing annotations should be overwritten

Reply:
Is it possible to reverse this behavior? In a system of ordered rules like this, you could have phantom blocking effects, e.g.
1. Rule 1 adds an entity: `{A B} C D`
2. Rule 2 replaces that entity with another `A {B C} D`
3. Rule 3 replaces that entity with another `A B {C D}`

The result is that Rule 1 fails to apply just because it conflicts with a rule that also didn't apply.

Reply:
The motivation behind this design is to be able to annotate entities from general to specific rules.

For example, say you want to annotate entities in the "date" domain in the HR assistant with the general rule to annotate sys_time and sys_date. However, for the "get_date_range_aggregate" you would also like to include "sys_interval" then you would a second rule specifying this intent and the additional entity. In this manner, the previous rule isn't overwritten but expanded upon at a more specific level.

An alternative to avoid any conflict is to write rules for each intent, however, this may quickly become very verbose and time consuming.